### PR TITLE
Introduce glnx_gen_temp_name()

### PR DESCRIPTION
**Pull request by cgwalters**
We have multiple copies growing again of this code.  glibc has this
API internally and uses it in multiple places, let's do the same.

### CODE REVIEW

**Comment by cgwalters** [comment]
Another consumer: https://github.com/ostreedev/ostree/pull/311

File: glnx-dirfd.c [hunk 1]
```diff
@@ -332,6 +323,31 @@ glnx_mkdtempat (int dfd,
       XXXXXX[4] = letters[v % NLETTERS];
       v /= NLETTERS;
       XXXXXX[5] = letters[v % NLETTERS];
+    }
+}
+
+/**
+ * glnx_mkdtempat:
+ * @dfd: Directory fd
+ * @tmpl: (type filename): template directory name
```
Comment (jlebon) [inline glnx-dirfd.c]: Could we also add ", the last 6 characters will be replaced" to the description here as well?

**Comment by cgwalters** [comment]
Sure, fixup :tokyo_tower: 

(BTW, I did audit the callers in rpm-ostree and they were all fine with this change)

**Comment by jlebon** [comment]
LGTM!

**Comment by cgwalters** [comment]
/me considers moving this to ostreedev and adding homu...at some point.

**Pull request closed without merging**